project/views.py

```diff
@@ -1,70 +1,180 @@
-                'created_at' : community.created_at,
-                'recomment'  : [{
-                    'user'       : recomment.user.fullname,
-                    'comment'    : recomment.comment,
-                    'created_at' : recomment.created_at
-                    } for recomment in community.community_set.all()
-                    ] if community.community_set.exists() else ''
-                } for community in project.community_set.filter(parent_id=None).order_by('-created_at')
-                ] if project.community_set.exists() else ''
-            }
-
-        return JsonResponse({
-            'project_info': project_info,
-            'creator_info': creator_info,
-            'tab'         : tab
-            }, status=200)
+from datetime         import datetime, timedelta
+
+from django.db        import transaction
+from django.views     import View
+from django.conf      import settings
+from django.http      import JsonResponse
+
+from my_settings      import (
+    SECRET_KEY, ALGORITHM,
+    AWS_ID    , AWS_KEY
+)
+
+from user.models      import User
+from user.utils       import login_decorator, user_decorator
+from user.models      import User
+from .models          import  (
+    Category, Project, Like,
+    Gift, Story, Community
+)
+     
+
+class RegisterView(View):
+    
+    @login_decorator
+    def post(self, request):
+        data    = json.loads(request.body)
+        user_id = request.user.id
+
+        try:
+            with transaction.atomic():
+                name          = data['name']
+                thumbnail_url = data['thumbnail_url']
+                summary       = data['summary']
+                category      = data['category']
+                story         = data['story']
+                goal_amount   = data['goal_amount']
+                total_amount  = data['total_amount']
+                opening_date  = data['opening_date']
+                closing_date  = data['closing_date']
+                project_uri   = data['project_uri']
+                gifts         = data['gifts']
+
+                category_id = Category.objects.get(name=category).id
+
+                Project.objects.create(
+                    category_id   = category_id,
+                    user_id       = user_id,
+                    name          = name,
+                    opening_date  = datetime.strptime(opening_date,'%Y-%m-%d'),
+                    closing_date  = datetime.strptime(closing_date,'%Y-%m-%d'),
+                    thumbnail_url = thumbnail_url,
+                    goal_amount   = goal_amount,
+                    total_amount  = total_amount,
+                    summary       = summary,
+                    project_uri   = project_uri
+                )
+                
+                project_id = Project.objects.get(name = name).id
+
+                Story.objects.create(
+                    content    = story,
+                    project_id = project_id
```

Comment:
```suggestion
                project = Project.objects.create(
                    category_id   = category_id,
                    user_id       = user_id,
                    name          = name,
                    opening_date  = datetime.strptime(opening_date,'%Y-%m-%d'),
                    closing_date  = datetime.strptime(closing_date,'%Y-%m-%d'),
                    thumbnail_url = thumbnail_url,
                    goal_amount   = goal_amount,
                    total_amount  = total_amount,
                    summary       = summary,
                    project_uri   = project_uri
                )
                
                Story.objects.create(
                    content    = story,
                    project    = project
```
project를 생성하고, 바로 생성한 project id가 필요한 경우라면 위와 같이 수정하면 좋을 것 같습니다.

Reply:
수정했습니다~!! 감사합니다!